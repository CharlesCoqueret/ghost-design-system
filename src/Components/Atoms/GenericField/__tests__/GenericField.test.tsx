import React from 'react';
import { GenericField } from '..';
import { render } from '@testing-library/react';

describe('GenericField Component', () => {
  it('GenericField renders', () => {
    const { container } = render(<GenericField />);

    expect(container).toMatchSnapshot();
  });

  it('GenericField renders with children', () => {
    const { container } = render(
      <GenericField>
        <div className='CHILD-TEST' />
      </GenericField>,
    );

    expect(container).toMatchSnapshot();
  });

  it('GenericField renders inline with error message, not displaying the error', () => {
    const { container } = render(<GenericField inline errorMessage='ERROR-MESSAGE' maxLength={10} />);

    expect(container).toMatchSnapshot();
  });

  it('GenericField renders with error message', () => {
    const { container } = render(<GenericField errorMessage='ERROR-MESSAGE' />);

    expect(container).toMatchSnapshot();
  });

  it('GenericField renders with error message in read only, not displaying the error', () => {
    const { container } = render(<GenericField readOnly errorMessage='ERROR-MESSAGE' maxLength={10} />);

    expect(container).toMatchSnapshot();
  });

  it('GenericField renders with error message in read only, not displaying the error, only displaying the helper text', () => {
    const { container } = render(
      <GenericField readOnly helperText='HELPER-TEXT' errorMessage='ERROR-MESSAGE' maxLength={10} />,
    );

    expect(container).toMatchSnapshot();
  });

  it('GenericField renders with error message and helper text, displaying only error', () => {
    const { container } = render(<GenericField helperText='HELPER-TEXT' errorMessage='ERROR-MESSAGE' maxLength={10} />);

    expect(container).toMatchSnapshot();
  });

  it('GenericField renders with helper text and counter', () => {
    const { container } = render(<GenericField helperText='HELPER-TEXT' inputLength={5} maxLength={10} />);

    expect(container).toMatchSnapshot();
  });

  it('GenericField renders with highlight in readonly', () => {
    const { container } = render(<GenericField readOnly highlighted />);

    expect(container).toMatchSnapshot();
  });

  it('GenericField renders with highlight not in readonly', () => {
    const { container } = render(<GenericField highlighted />);

    expect(container).toMatchSnapshot();
  });

  it('GenericField renders with invertInputDescription', () => {
    const { container, rerender } = render(
      <GenericField helperText='HELPER' invertInputDescription>
        CHILD
      </GenericField>,
    );

    expect(container).toMatchSnapshot();

    rerender(<GenericField helperText='HELPER'>CHILD</GenericField>);
    expect(container).toMatchSnapshot();
  });
});
