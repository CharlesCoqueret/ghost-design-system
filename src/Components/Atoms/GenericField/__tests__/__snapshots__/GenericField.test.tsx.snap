// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GenericField Component GenericField renders 1`] = `
<div>
  <div
    class="fieldGroup"
  />
</div>
`;

exports[`GenericField Component GenericField renders inline with error message, not displaying the error 1`] = `
<div>
  <div
    class="fieldGroup inline"
  />
</div>
`;

exports[`GenericField Component GenericField renders with children 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <div
      class="CHILD-TEST"
    />
  </div>
</div>
`;

exports[`GenericField Component GenericField renders with error message 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <div
      class="message"
    >
      <div
        class="error field-error"
      >
        ERROR-MESSAGE
      </div>
    </div>
  </div>
</div>
`;

exports[`GenericField Component GenericField renders with error message and helper text, displaying only error 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <div
      class="message"
    >
      <div
        class="error field-error"
      >
        ERROR-MESSAGE
      </div>
    </div>
  </div>
</div>
`;

exports[`GenericField Component GenericField renders with error message in read only, not displaying the error 1`] = `
<div>
  <div
    class="fieldGroup"
  />
</div>
`;

exports[`GenericField Component GenericField renders with error message in read only, not displaying the error, only displaying the helper text 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <div
      class="message"
    >
      <div
        class="helper"
      >
        HELPER-TEXT
      </div>
    </div>
  </div>
</div>
`;

exports[`GenericField Component GenericField renders with helper text and counter 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <div
      class="message"
    >
      <div
        class="helper"
      >
        HELPER-TEXT
      </div>
      <div
        class="counter"
      >
        5 / 10
      </div>
    </div>
  </div>
</div>
`;

exports[`GenericField Component GenericField renders with highlight in readonly 1`] = `
<div>
  <div
    class="fieldGroup"
  />
</div>
`;

exports[`GenericField Component GenericField renders with highlight not in readonly 1`] = `
<div>
  <div
    class="fieldGroup"
  />
</div>
`;

exports[`GenericField Component GenericField renders with invertInputDescription 1`] = `
<div>
  <div
    class="fieldGroup"
  >
    <div
      class="message inverted"
    >
      <div
        class="helper"
      >
        HELPER
      </div>
    </div>
    CHILD
  </div>
</div>
`;

exports[`GenericField Component GenericField renders with invertInputDescription 2`] = `
<div>
  <div
    class="fieldGroup"
  >
    CHILD
    <div
      class="message"
    >
      <div
        class="helper"
      >
        HELPER
      </div>
    </div>
  </div>
</div>
`;
